include/oneapi/tbb/blocked_rangeNd.h

```diff
@@ -60,23 +57,36 @@ class blocked_rangeNd_impl<Value, N, detail::index_sequence<Is...>> {
     //! Type of a value.
     using value_type = Value;
 
-private:
     //! Helper type to construct range with N tbb::blocked_range<value_type> objects.
     template<std::size_t>
-    using dim_type_helper = tbb::blocked_range<value_type>;
+    using dim_range_type = tbb::blocked_range<value_type>;
 
-public:
     blocked_rangeNd_impl() = delete;
 
     //! Constructs N-dimensional range over N half-open intervals each represented as tbb::blocked_range<Value>.
-    blocked_rangeNd_impl(const dim_type_helper<Is>&... args) : my_dims{ {args...} } {}
+    blocked_rangeNd_impl(const dim_range_type<Is>&... args) : my_dims{ {args...} } {
+      __TBB_ASSERT(my_dims.size()==N, "range is not properly initialised" );
+    }
+
+    blocked_rangeNd_impl(
+      Value begin[N],
+      Value end[N],
+      typename tbb::blocked_range<value_type>::size_type grainsize
+    ):
+      my_dims() {
```

Comment:
The following way to do the initialization of `my_dims` should I think work:
```suggestion
    blocked_rangeNd_impl(
      Value begin[N],
      Value end[N],
      typename tbb::blocked_range<value_type>::size_type grainsize
    ):
      my_dims{ tbb::blocked_range<value_type>{begin[Is], end[Is], grainsize}... } {
```
And then the body can be simplified as there is no need to initialize data elements.